consensus/core/src/block_verifier.rs

```diff
@@ -284,22 +297,37 @@ mod test {
             Ok(())
         }
 
-        async fn verify_and_vote_batch(
+        // Fails verification if any transaction is < 4 bytes.
+        // Rejects transactions with length [4, 16) bytes.
+        fn verify_and_vote_batch(
```

Comment:
I know this is just a test method, but in practice when transactions rejected will only be returned if they are not certified. Transactions that fail validation would cause the whole block to be rejected or will those be sent as rejected transactions?

If individual transactions can fail validation and get returned as a rejected transaction then can you update the comment above verify_and_vote_batch in transaction.rs to specifically say block validation

Reply:
Updated the comments. Basically, verification / validation errors mean the block author is equivocating. The whole block gets rejected and there is no need to vote. Client should retry submission on different authorities if it cannot collect enough info from a quorum after awhile, in case the initial submission authority is equivocating or just unable to publish blocks. Transactions are rejected only through voting on transactions in verified blocks.